docs/user-guide/jobs.md

```diff
@@ -192,6 +192,15 @@ A single Job object can also be used to control multiple pods running in
 parallel.  There are several different [patterns for running parallel
 jobs](#job-patterns).
 
+Leaving completions unset and setting it to 1 are different.
+When it is set to 1, then parallelism cannot be scaled above 1.
+When it is unset, parallelism can be scaled to any number, and multiple pods may run in parallel.
+When it is unset, _any_ pod that signals its own success (by exiting with exit code 0) also signals
```

Comment:
It would be good to stress here what you wrote in the controller code, which is:

Each pod *should be* capable of determining whether or not the entire Job is done.  Subsequent pods are not expected to fail, but if they do, the failure is ignored.  Once any pod succeeds, the controller waits for remaining pods to finish, and then the job is complete.